controllers/machineset_controller.go

```diff
@@ -589,3 +582,115 @@ func (r *MachineSetReconciler) hasMatchingLabels(machineSet *clusterv1.MachineSe
 func (r *MachineSetReconciler) shouldAdopt(ms *clusterv1.MachineSet) bool {
 	return !util.HasOwner(ms.OwnerReferences, clusterv1.GroupVersion.String(), []string{"MachineDeployment", "Cluster"})
 }
+
+func (r *MachineSetReconciler) calculateStatus(cluster *clusterv1.Cluster, ms *clusterv1.MachineSet, filteredMachines []*clusterv1.Machine) clusterv1.MachineSetStatus {
+	logger := r.Log.WithValues("machineset", ms.Name, "namespace", ms.Namespace)
+	newStatus := ms.Status
+
+	// Copy label selector to its status counterpart in string format.
+	// This is necessary for CRDs including scale subresources.
+	selector, _ := metav1.LabelSelectorAsSelector(&ms.Spec.Selector)
```

Comment:
Is it actually safe to ignore the error here? I know we have a validating webhook now, but that wouldn't affect pre-existing stored resources (unless they have been read and then re-written to the new storage version).

Reply:
I figured given that we're coming from a `LabelSelector` type that this function should never return an error, unless something very wrong happened.

I'll add it though, just to be safe